features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/GatewayEnvironments/AddEditVhost.jsx

```diff
@@ -0,0 +1,282 @@
+        setUserVhosts(tempItems);
+    };
+    const changeHandler = (field) => ({ target: { name, value } }) => {
+        let theValue = value;
+        if (field === 'httpContext') {
+            // remove slashes in start and end of httpContext
+            theValue = value.replace(/(^\/|\/$)/g, '');
+        }
+        updateChanges(name, field, theValue);
+    };
+
+    const [openRemoveVhost, setOpenRemoveVhost] = useState(false);
+    const [selectedKey, setSelectedKey] = useState(0);
+
+    const handleRemoveVhost = (key) => {
+        const theKey = key || selectedKey;
+        const tempItems = userVhosts.filter((vhost) => vhost.key !== theKey);
+        setUserVhosts(tempItems);
+        setOpenRemoveVhost(false);
+    };
+
+    const handleRemoveVhostConfirm = (key, isNew) => {
+        if (isNew) { // not already existing one
+            handleRemoveVhost(key);
+        } else { // already existing vhost
+            setSelectedKey(key);
+            setOpenRemoveVhost(true);
+        }
+    };
+
+    const handleClose = () => {
+        setOpenRemoveVhost(false);
+    };
+
+    const handleNewVhost = () => {
+        const vhost = defaultVhost;
+        vhost.key = '' + id;
+        setId(id + 1);
+        const tempItems = [...userVhosts, vhost];
+        setUserVhosts(tempItems);
+    };
+
+    useEffect(() => {
+        const nonEmptyItems = [];
+        for (const userVhost of userVhosts) {
+            if (userVhost.host && userVhost.host.trim() !== '') {
+                nonEmptyItems.push(userVhost);
+            }
+        }
+        onVhostChange({ target: { name: 'vhosts', value: nonEmptyItems } });
+    }, [userVhosts]);
+
+    useEffect(() => {
+        if (initialVhosts && initialVhosts.length > 0) {
+            let i = 0;
+            setUserVhosts(initialVhosts.map((vhost) => {
+                const keyedVhost = vhost;
+                keyedVhost.key = '' + i++;
+                return keyedVhost;
+            }));
+            setId(i);
+        } else {
+            setId(id + 1);
+            const vhost = defaultVhost;
+            vhost.key = '' + id;
+            setUserVhosts([vhost]);
+        }
+    }, []);
+
+    let vhostCounter = 1;
+    return (
+        <FormGroup>
+            <Grid container spacing={0}>
+                {userVhosts && userVhosts.map((vhost) => (
+                    <Grid item xs={12}>
+                        <Paper variant='outlined' className={classes.vhostPaper}>
+                            <Grid container direction='row' spacing={0}>
+                                {/* VHost's host name */}
+                                <Grid item xs={9}>
+                                    <TextField
+                                        margin='dense'
+                                        name={vhost.key}
+                                        disabled={!vhost.isNew}
+                                        onChange={changeHandler('host')}
```

Comment:
This will be always 2 right ?

Reply:
because of 
`let vhostCounter = 1;`

Reply:
Nope, it is changing, since it is `vhostCounter++`
In the UI Vhost number is incremented,
![image](https://user-images.githubusercontent.com/23183042/108792342-09d6ea80-75a7-11eb-967c-8bc1d67d7ac0.png)